File: PetSocietyWeb/PetSociety.API/DTOs/User/FavoriteDto.cs
namespace PetSociety.API.DTOs.User
{
        // 用於 "新增" 收藏 (前端傳給後端的)
        public class AddFavoriteDto
        {
            public int TargetId { get; set; }
            public int TargetType { get; set; }
            public string Title { get; set; }
            public string? ImageUrl { get; set; }
            public decimal? Price { get; set; }
            public string? Intro { get; set; }
        }

        // 用於 "顯示" 收藏列表 (後端傳給前端的)
        public class FavoriteItemDto
        {
            public int FavoriteId { get; set; }
            public int TargetId { get; set; }
            public int TargetType { get; set; }
            public string Title { get; set; }
            public string? ImageUrl { get; set; }
            public decimal? Price { get; set; }
            public string? Intro { get; set; }
            public string CreateDate { get; set; } // 格式化後的日期字串
        }
}

File: petfront/PetSocietyFrontWeb/src/app/features/member/Interfaces/favorite.dto.ts
// 定義收藏類型的 Enum (對應資料庫的 TargetType)
export enum FavoriteType {
  Product = 1,
  Activity = 2,
  Course = 3,
  Article = 4
}

// 收藏項目的 DTO (用來顯示列表)
export interface FavoriteItemDto {
  favoriteId: number;
  targetId: number;
  targetType: number;

  // 快照欄位
  title: string;
  imageUrl: string;
  description?: string;
  price?: number;
  date?: string;
}
